src/github.com/matrix-org/dendrite/syncapi/consumers/roomserver.go

```diff
@@ -177,6 +202,44 @@ func (s *OutputRoomEvent) lookupStateEvents(
 	return result, nil
 }
 
+func (s *OutputRoomEvent) updateMemberEvent(
+	event gomatrixserverlib.Event, keyID gomatrixserverlib.KeyID,
+	privateKey []byte,
+) (gomatrixserverlib.Event, error) {
+	if event.Type() != "m.room.member" {
+		return event, nil
+	}
+	membership, err := event.Membership()
+	if err != nil {
+		return event, err
+	}
+	if membership != "join" {
+		return event, nil
+	}
```

Comment:
Should we be settting the "prev_content", "replaces_state" and "prev_sender" for every state event? not just the join m.room.member events?